src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -801,8 +801,9 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             bool hasNaturalType = true;
 
             var boundArguments = ArrayBuilder<BoundExpression>.GetInstance(arguments.Count);
-            var elementTypes = ArrayBuilder<TypeSymbol>.GetInstance(arguments.Count);
+            var elementTypes = ArrayBuilder<TypeSymbolWithAnnotations>.GetInstance(arguments.Count);
             var elementLocations = ArrayBuilder<Location>.GetInstance(arguments.Count);
+            bool includeNullability = Compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticNullChecking);
```

Comment:
Do we need to condition logic here on feature being enabled? Could we always track the nullability (even in C# 7), just not using it when it matters?

Reply:
Added `// PROTOTYPE` comment.